frontend/src/container/NewWidget/index.tsx

```diff
@@ -331,10 +356,39 @@ function NewWidget({ selectedGraph }: NewWidgetProps): JSX.Element {
 				centered
 				open={saveModal}
 				width={600}
+			>
+				{!isQueryModified ? (
+					<Typography>
+						Your graph built with <QueryTypeTag queryType={currentQuery.queryType} />{' '}
+						query will be saved. Press OK to confirm.
+					</Typography>
+				) : (
+					<Typography>
+						There are unsaved changes in the Query builder, please stage and run the
+						query or the changes will be lost. Press OK to discard.
+					</Typography>
+				)}
+			</Modal>
+			<Modal
+				title={
+					<Space>
+						<WarningOutlined style={{ fontSize: '16px', color: '#fdd600' }} />
+						Unsaved Changes
+					</Space>
+				}
+				focusTriggerAfterClose
+				forceRender
+				destroyOnClose
+				closable
+				onCancel={(): void => setDiscardModal(false)}
```

Comment:
This function will be recreated on every re-render. Can we not use a toggle function for this ?